azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java

```diff
@@ -47,6 +48,82 @@ public ExecutableFlow fetchExecutableFlow(int execId)
       String flowContains, String userNameContains, int status, long startData,
       long endData, int skip, int num) throws ExecutorManagerException;
 
+  /**
+   * Fetch all executors from executors table with active = true
+   *
+   * @return List<Executor>
+   * @throws ExecutorManagerException
+   */
+  public List<Executor> fetchActiveExecutors() throws ExecutorManagerException;
+
+  /**
+   * Fetch executor from executors with a given (host, port)
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor fetchExecutor(String host, int port)
+    throws ExecutorManagerException;
+
+  /**
+   * Fetch executor from executors with a given executorId
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor fetchExecutor(int executorId) throws ExecutorManagerException;
+
+  /**
+   * create an executor and insert in executors table. Fails, if a executor with
+   * (host, port) already exist
+   *
+   * @return Executor
+   * @throws ExecutorManagerException
+   */
+  public Executor addExecutor(String host, int port)
+    throws ExecutorManagerException;
+
+  /**
+   * inactivate an executor in executors table
+   *
+   * @param executorId
+   * @throws ExecutorManagerException
+   */
+  public void inactivateExecutor(int executorId)
```

Comment:
Does it make sense to combine these two methods into updateExecutor(Executor executor) method?